src/diffusers/quantizers/modelopt/modelopt_quantizer.py

```diff
@@ -0,0 +1,163 @@
+    import torch
+
+if is_accelerate_available():
+    from accelerate.utils import set_module_tensor_to_device
+
+
+logger = logging.get_logger(__name__)
+
+
+class NVIDIAModelOptQuantizer(DiffusersQuantizer):
+    r"""
+    Diffusers Quantizer for TensorRT Model Optimizer
+    """
+
+    use_keep_in_fp32_modules = True
+    requires_calibration = False
+    required_packages = ["modelopt"]
+
+    def __init__(self, quantization_config, **kwargs):
+        super().__init__(quantization_config, **kwargs)
+
+    def validate_environment(self, *args, **kwargs):
+        if not is_nvidia_modelopt_available():
+            raise ImportError(
+                "Loading an nvidia-modelopt quantized model requires nvidia-modelopt library (`pip install nvidia-modelopt`)"
+            )
+        if not is_nvidia_modelopt_version(">=", "0.25.0"):
+            raise ImportError(
+                "Loading an nvidia-modelopt quantized model requires `nvidia-modelopt>=0.25.0`. "
+                "Please upgrade your installation with `pip install --upgrade nvidia-modelopt"
+            )
+
+        self.offload = False
+
+        device_map = kwargs.get("device_map", None)
+        if isinstance(device_map, dict):
+            if "cpu" in device_map.values() or "disk" in device_map.values():
+                if self.pre_quantized:
+                    raise ValueError(
+                        "You are attempting to perform cpu/disk offload with a pre-quantized modelopt model "
+                        "This is not supported yet. Please remove the CPU or disk device from the `device_map` argument."
+                    )
+                else:
+                    self.offload = True
+
+    def check_if_quantized_param(
+        self,
+        model: "ModelMixin",
+        param_value: "torch.Tensor",
+        param_name: str,
+        state_dict: Dict[str, Any],
+        **kwargs,
+    ):
+        # ModelOpt imports diffusers internally. This is here to prevent circular imports
+        from modelopt.torch.quantization.nn import QuantInputBase, SequentialQuantizer, TensorQuantizer
+        from modelopt.torch.quantization.qtensor import BaseQuantizedTensor
+
+        def is_param_quantized(module):
+            for _module in module.modules():
+                if isinstance(_module, TensorQuantizer) and not _module._dequantize:
+                    return True
+                elif isinstance(_module, SequentialQuantizer):
+                    for q in _module:
+                        if isinstance(q, TensorQuantizer) and not q._dequantize:
+                            return True
+            return False
+
+        module, tensor_name = get_module_from_name(model, param_name)
+        if self.pre_quantized and any(isinstance(module, t) for t in [BaseQuantizedTensor]):
+            return True
+        elif isinstance(module, QuantInputBase) and "weight" in tensor_name:
+            return is_param_quantized(module)
+        return False
```

Comment:
Why dont we simply check if the model has any `TensorQuantizer`?  `is_quantized` API from `modelopt` does exactly this.
```suggestion
        from modelopt.torch.quantization.utils import is_quantized
        return is_quantized(model)
```

Reply:
using `is_quantized` would mean the following

- won't be able to check quantization for `SequentialQuantizer`
- for real_quantizer we need to also check if `_dequantize` is True (this means that it is actually compressed and not just replacing layers with their counterparts).

hence this function, let me know if this looks correct with above two cases in mind

Reply:
> won't be able to check quantization for SequentialQuantizer
SequentialQuantizer is an `nn.Sequential` container for `TensorQuantizer`. See https://github.com/NVIDIA/TensorRT-Model-Optimizer/blob/e048fb239de4e4cb6a5f8f86ba70455478a1847e/modelopt/torch/quantization/nn/modules/tensor_quantizer.py#L1174

Reply:
We should be able to verify if `is_quantized(model)` cuts the deal for us minimally. @ishan-modi WDYT?

Reply:
Agreed, I will confirm and make the change !